pkg/limits/frontend/service.go

```diff
@@ -77,24 +91,65 @@ func (s *RingIngestLimitsService) forGivenReplicaSet(ctx context.Context, replic
 	return responses, nil
 }
 
-func (s *RingIngestLimitsService) ExceedsLimits(ctx context.Context, r *logproto.ExceedsLimitsRequest) (*logproto.ExceedsLimitsResponse, error) {
-	req := &logproto.GetStreamUsageRequest{
-		Tenant: r.Tenant,
-	}
-	resps, err := s.forAllBackends(ctx, func(_ context.Context, client logproto.IngestLimitsClient) (interface{}, error) {
-		return client.GetStreamUsage(ctx, req)
+func (s *RingIngestLimitsService) ExceedsLimits(ctx context.Context, req *logproto.ExceedsLimitsRequest) (*logproto.ExceedsLimitsResponse, error) {
+	resps, err := s.forAllBackends(ctx, func(_ context.Context, client logproto.IngestLimitsClient) (*logproto.GetStreamUsageResponse, error) {
+		return client.GetStreamUsage(ctx, &logproto.GetStreamUsageRequest{
+			Tenant: req.Tenant,
+		})
 	})
 	if err != nil {
 		return nil, err
 	}
-	var sum uint64
+
+	maxGlobalStreams := s.limits.MaxGlobalStreamsPerUser(req.Tenant)
+
+	var (
+		activeStreamsTotal uint64
+		uniqueStreamHashes = make(map[uint64]bool)
+	)
 	for _, resp := range resps {
-		sum += resp.Response.(*logproto.GetStreamUsageResponse).ActiveStreams
+		if resp.Response == nil {
+			continue
+		}
+
+		activeStreamsTotal += resp.Response.ActiveStreams
+
+		// Record the unique stream hashes
+		// and decrement the active streams
+		// total if the stream hash is already
+		// recorded
+		for _, stream := range resp.Response.RecordedStreams {
+			if !uniqueStreamHashes[stream.StreamHash] {
+				uniqueStreamHashes[stream.StreamHash] = true
```

Comment:
I find the code easier to read if we increment unique streams inside the if statement, instead of add the total and then subtract the duplicates. Is the latter much faster?